adapter/internal/oasparser/envoyconf/envoyconf_internal_test.go

```diff
@@ -250,136 +238,136 @@ func TestGenerateTLSCert(t *testing.T) {
+			pathMatchType: gwapiv1b1.PathMatchRegularExpression,
+			resourcePath:  "/pet/[\\d]{1}",
+			userInputPath: "/pet/5",
+			message:       "when regex is provided end of the path",
 			isMatched:     true,
 		},
 		{
-			basePath:      `/v2`,
-			resourcePath:  `/pet/{petId}`,
-			userInputPath: "/v2/pet/5/",
+			pathMatchType: gwapiv1b1.PathMatchRegularExpression,
+			resourcePath:  "/pet/[\\d]{1}",
+			userInputPath: "/pet/5/",
 			message:       "when the input path does not have tailing slash and user input path has trailing slash",
 			isMatched:     true,
 		},
 		{
-			basePath:      "/v2",
-			resourcePath:  "/pet/{petId}/",
-			userInputPath: "/v2/pet/5",
+			pathMatchType: gwapiv1b1.PathMatchRegularExpression,
+			resourcePath:  "/pet/[\\d]{1}/",
+			userInputPath: "/pet/5",
 			message:       "when the input path has tailing slash and user input path does not have trailing slash",
 			isMatched:     true,
 		},
 		{
-			basePath:      "/v2",
-			resourcePath:  "/pet/{petId}/",
-			userInputPath: "/v2/pet/5/",
+			pathMatchType: gwapiv1b1.PathMatchRegularExpression,
+			resourcePath:  "/pet/[\\d]{1}/",
+			userInputPath: "/pet/5/",
 			message:       "when both the input path and user input path has trailing slash",
 			isMatched:     true,
 		},
 		{
-			basePath:      "/v2",
-			resourcePath:  "/pet/{petId}/info",
-			userInputPath: "/v2/pet/5/info",
-			message:       "when path parameter is provided in the middle of the path",
-			isMatched:     true,
-		},
-		{
-			basePath:      "/v2",
-			resourcePath:  "/pet/{petId}",
-			userInputPath: "/v2/pet/5",
-			message:       "when path parameter is provided end of the path",
+			pathMatchType: gwapiv1b1.PathMatchRegularExpression,
+			resourcePath:  "/pet/[\\d]{1}/info",
+			userInputPath: "/pet/5/info",
+			message:       "when regex is provided in the middle of the path",
 			isMatched:     true,
 		},
 		{
-			basePath:      "/v2",
-			resourcePath:  "/pet/{petId}/tst/{petId}",
-			userInputPath: "/v2/pet/5/tst/3",
-			message:       "when multiple path parameter are provided",
+			pathMatchType: gwapiv1b1.PathMatchRegularExpression,
+			resourcePath:  "/pet/[\\d]{1}/tst/[\\d]{1}",
+			userInputPath: "/pet/5/tst/3",
+			message:       "when multiple regex match sections are provided",
 			isMatched:     true,
 		},
 		{
-			basePath:      "/v2",
-			resourcePath:  "/pet/{petId}",
-			userInputPath: "/v2/pet/5/test",
+			pathMatchType: gwapiv1b1.PathMatchRegularExpression,
+			resourcePath:  "/pet/[\\d]{1}",
+			userInputPath: "/pet/5/test",
 			message:       "when path parameter is provided end of the path and provide incorrect path",
 			isMatched:     false,
 		},
 		{
-			basePath:      "/v2",
+			pathMatchType: gwapiv1b1.PathMatchExact,
 			resourcePath:  "/pet/5",
-			userInputPath: "/v2/pett/5",
-			message:       "when provide a incorrect path",
+			userInputPath: "/pet/5",
+			message:       "when using an exact match type",
+			isMatched:     true,
+		},
+		{
```

Comment:
shall we add a test for when there's trailing slash when exact match

Reply:
add trailing slash test for all cases

Reply:
added more tests